orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -217,129 +235,225 @@ class JedisExecutionRepository implements ExecutionRepository {
 
   @Override
   void storeStage(OrchestrationStage stage) {
-    withJedis { Jedis jedis ->
+    withJedis(getJedisPoolForId("orchestration:${stage.execution.id}")) { Jedis jedis ->
       storeStageInternal(jedis, Orchestration, stage)
     }
   }
 
   @Override
   Pipeline retrievePipeline(String id) {
-    withJedis { Jedis jedis ->
+    withJedis(getJedisPoolForId("pipeline:${id}")) { Jedis jedis ->
       retrieveInternal(jedis, Pipeline, id)
     }
   }
 
   @Override
   void deletePipeline(String id) {
-    withJedis { Jedis jedis ->
+    withJedis(getJedisPoolForId("pipeline:${id}")) { Jedis jedis ->
       deleteInternal(jedis, Pipeline, id)
     }
   }
 
   @Override
   Observable<Pipeline> retrievePipelines() {
-    all(Pipeline)
+    def observables = [all(Pipeline, jedisPool)]
+    jedisPoolPrevious.map { jedisPool -> observables << all(Pipeline, jedisPool) }
+    return Observable.merge(observables)
   }
 
   @Override
   Observable<Pipeline> retrievePipelinesForApplication(String application) {
-    allForApplication(Pipeline, application)
+    def observables = [allForApplication(Pipeline, application, jedisPool)]
+    jedisPoolPrevious.map { jedisPool -> observables << allForApplication(Pipeline, application, jedisPool) }
+    return Observable.merge(observables)
   }
 
   @Override
   @CompileDynamic
   Observable<Pipeline> retrievePipelinesForPipelineConfigId(String pipelineConfigId,
                                                             ExecutionCriteria criteria) {
-    def filteredPipelineIds = null as List<String>
+    /**
+     * Fetch pipeline ids from the primary redis (and secondary if configured)
+     */
+    Map<Pool<Jedis>, List<String>> filteredPipelineIdsByJedis = [:].withDefault { [] }
     if (criteria.statuses) {
-      filteredPipelineIds = []
-      withJedis { Jedis jedis ->
-        def pipelineKeys = jedis.zrevrange(executionsByPipelineKey(pipelineConfigId), 0, -1)
-        def allowedExecutionStatuses = criteria.statuses*.toString() as Set<String>
-
-        def pipeline = jedis.pipelined()
-        def fetches = pipelineKeys.collect {
-          pipeline.hget("pipeline:${it}" as String, "status")
-        }
-        pipeline.sync()
+      ([jedisPool] + (jedisPoolPrevious.present ? [jedisPoolPrevious.get()] : [])).each { Pool<Jedis> jedisPool ->
```

Comment:
You're shadowing a field name there which is a little confusing at first glance at the code below